### PR TITLE
Add "OpenGL Profile" setting with restored support for GL2.1

### PR DESCRIPTION
**Pull request by pchote**
As discussed on IRC and Discord, this PR temporarily restores the OpenGL 2.1 support removed by #17010 as an optional (and disabled by default) rendering mode that can be used by the few people still suffering from GPU driver issues. This will need to be removed again in the future, but hopefully at that point we will be able to support Direct X (either directly or via ANGLE) as an alternative.

### CODE REVIEW

File: OpenRA.Platforms.Default/OpenGL.cs [hunk 1]
```diff
@@ -594,16 +591,40 @@ public static void Initialize()
...
-				glDeleteRenderbuffers = Bind<DeleteRenderbuffers>("glDeleteRenderbuffers");
-				glFramebufferRenderbuffer = Bind<FramebufferRenderbuffer>("glFramebufferRenderbuffer");
-				glCheckFramebufferStatus = Bind<CheckFramebufferStatus>("glCheckFramebufferStatus");
+
+				if (Profile != GLProfile.Legacy)
+				{
+					glGenVertexArrays = Bind<GenVertexArrays>("glGenVertexArrays");
+					glBindVertexArray = Bind<BindVertexArray>("glBindVertexArray");
+					glBindFragDataLocation = Bind<BindFragDataLocation>("glBindFragDataLocation");
+
```
Comment (reaperrr) [inline OpenRA.Platforms.Default/OpenGL.cs]: Minor nit: this empty line looks a bit arbitrary.

File: OpenRA.Platforms.Default/OpenGL.cs [hunk 2]
```diff
@@ -594,16 +591,40 @@ public static void Initialize()
...
+					glDeleteRenderbuffers = Bind<DeleteRenderbuffers>("glDeleteRenderbuffers");
+					glFramebufferRenderbuffer = Bind<FramebufferRenderbuffer>("glFramebufferRenderbuffer");
+					glCheckFramebufferStatus = Bind<CheckFramebufferStatus>("glCheckFramebufferStatus");
+				}
+				else
+				{
+					glGenVertexArrays = null;
+					glBindVertexArray = null;
+					glBindFragDataLocation = null;
+					glGenFramebuffers = Bind<GenFramebuffers>("glGenFramebuffersEXT");
```
Comment (chrisforbes) [inline OpenRA.Platforms.Default/OpenGL.cs:624]: ARB_ version? it has the same semantics as core. EXT is slightly different

Comment (pchote) [reply OpenRA.Platforms.Default/OpenGL.cs:624]: EXT is what we were using for years before ripping it out, so i'm hesitant to risk changing it now!

**Comment by pchote** [comment]
Updated:
* Fixed style nit
* Changed debug message blacklist to more specifically target @cjshmyr's broken driver

File: OpenRA.Platforms.Default/OpenGL.cs (hunk 2 above)
Comment (chrisforbes) [reply OpenRA.Platforms.Default/OpenGL.cs:624]: OK then

**Review by chrisforbes — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/d11a9744f53fbfbb725d21f3e4111fce32250edb...363d580329bcb0b29d59f818ecd9400e3afc13e8)

https://github.com/OpenRA/OpenRA/commit/390fbe671824529c03359cdbfdec5c6d96f4a846
https://github.com/OpenRA/OpenRA/commit/835a9e605ab689b5347e1733de3d9e5f3a479301
https://github.com/OpenRA/OpenRA/commit/5b6b4ab4ee6d4e184f1ceb71109772f1716c933d
https://github.com/OpenRA/OpenRA/commit/426c3d68a1f2a61a66b53bba046023bd79b1aa55
https://github.com/OpenRA/OpenRA/commit/c13e078ebd573c9cfebc5aa29690fa524a5d0579

**Pull request merged**